packages/vendure-plugin-invoices/src/ui/util.ts

```diff
@@ -0,0 +1,75 @@
+    id: isStyled ? 'regenerate-invoice-styled' : 'regenerate-invoice',
+    label: 'Regenerate Invoice',
+    locationId: 'order-detail',
+    requiresPermission: ['AllowInvoicesPermission'],
+    buttonColor: isStyled ? 'warning' : 'primary',
+    buttonStyle: isStyled ? 'solid' : 'outline',
+    onClick: (event, context) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const orderId = context.route.snapshot.params['id'];
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      (event.target as HTMLButtonElement).disabled = true;
+      context.dataService
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
+        .mutate(createInvoice, { orderId }, (cache: ApolloCache<any>) => {
+          cache.evict({ fieldName: 'invoices', id: `Order:${orderId}` });
+        })
+        .pipe(
+          catchError((error) => {
+            context.notificationService.error('Failed to regenerate invoices');
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            (event.target as HTMLButtonElement).disabled = false;
+            throw new Error(JSON.stringify(error));
+          })
+        )
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .subscribe((data: any) => {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          if (data?.createInvoice?.id) {
+            context.notificationService.success(
+              'Invoice has been regenerated successfully'
+            );
+          } else {
+            context.notificationService.error('Failed to regenerate invoices');
+          }
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          (event.target as HTMLButtonElement).disabled = false;
+        });
+    },
+    buttonState: (context) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const orderId = context.route.snapshot.params['id'];
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment
+      const order$ = context.dataService.query<any, GetOrderQueryVariables>(
+        GET_ORDER,
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        { id: orderId }
+      );
+      return order$.stream$.pipe(
+        map(({ order }) => {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+          const latestInvoice = order?.invoices[0];
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          const orderTotalMatches =
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            latestInvoice.orderTotals.totalWithTax === order?.totalWithTax;
+          const showButton = isStyled !== orderTotalMatches;
```

Comment:
This seems a bit hard to read. Shouldn't it be something like `showButton = isStyled && !orderTotalMatches` (reading as 'show button if isStyled and order total doesn't match').

Also, can we refactor `isStyled` to something like `isWarningButton`? Is styled is a bit ambiguous